### PR TITLE
upgrade nix -> 0.27.1, address kqueue EINTR issue

### PR DESCRIPTION
**Pull request by elliptic-kitty**
This PR upgrade nix to v0.27.1

In the same time also address #5

### CODE REVIEW

File: freebsd/src/event.rs [hunk 1]
```diff
@@ -149,6 +151,58 @@ impl std::future::Future for EventFdNotified<'_> {
...
+                std::ptr::null(),
+            )
+        };
+
+        let errno = nix::errno::errno();
+
+        match res {
+            -1 => {
+                if errno != nix::libc::EINTR {
+                    return Err(nix::errno::Errno::EINTR);
```
Comment (Freaky) [inline freebsd/src/event.rs]: This should probably be `Err(errno)`, rather than returning the one error it can't be.

File: freebsd/src/event.rs [hunk 2]
```diff
@@ -149,6 +151,58 @@ impl std::future::Future for EventFdNotified<'_> {
...
+    fn wait_events(&self, changelist: &[KEvent], eventlist: &mut [KEvent]) -> nix::Result<usize>;
+}
+
+pub fn kevent_classic(
+    kq: i32,
+    changelist: &[KEvent],
+    eventlist: &mut [KEvent],
+) -> nix::Result<usize> {
+    let mut interrupted = true;
+    while interrupted {
```
Comment (Freaky) [inline freebsd/src/event.rs]: At no point is `interrupted` false.  Just use `loop {}`.

File: freebsd/src/event.rs [hunk 3]
```diff
@@ -149,6 +151,58 @@ impl std::future::Future for EventFdNotified<'_> {
...
+            size => return Ok(size as usize),
+        }
+    }
+    unreachable!()
+}
+
+impl KqueueExt for nix::sys::event::Kqueue {
+    fn wait_events(&self, changelist: &[KEvent], eventlist: &mut [KEvent]) -> nix::Result<usize> {
+        let mut interrupted = true;
+        while interrupted {
```
Comment (Freaky) [inline freebsd/src/event.rs]: As before, just `loop {}` is sufficient.

File: xc-bin/src/jailfile/directives/run.rs [hunk 4]
```diff
@@ -119,7 +118,7 @@ impl Directive for RunDirective {
         let (stderr_a, stderr_b) = pipe()?;
         let (stdin_a, stdin_b) = pipe()?;
 
-        let kq = unsafe { nix::libc::kqueue() };
+        let kq = unsafe { freebsd::nix::libc::kqueue() };
```
Comment (daniloegea) [inline xc-bin/src/jailfile/directives/run.rs:121]: Is there any particular reason to continue to use raw kqueue file descriptors?

File: freebsd/src/event.rs (hunk 1 above)
Comment (elliptic-kitty) [reply freebsd/src/event.rs]: That is indeed a typo

File: xc-bin/src/jailfile/directives/run.rs (hunk 4 above)
Comment (elliptic-kitty) [reply xc-bin/src/jailfile/directives/run.rs:121]: there are places where using raw file descriptors helps dealing with the borrow checker. If this particular case I am just trying to no risk breaking existing code

**Comment by khng300** [comment]
lgtm.

**Pull request merged**